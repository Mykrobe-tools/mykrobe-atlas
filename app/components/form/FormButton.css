.wrap {
  composes: wrap from "css/form.css";
}

.input {
  composes: buttonPrimary from "css/controls.css";
  cursor: pointer;
}
