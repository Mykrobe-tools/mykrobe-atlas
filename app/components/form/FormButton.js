/* @flow */

import React, { Component, PropTypes } from 'react';

import styles from './FormButton.css';

class FormButton extends Component {

  onClick(event: Event) {
    const {onClick} = this.props;
    if (onClick) {
      event.preventDefault();
      onClick(event);
    }
  }

  render() {
    const {label, type} = this.props;
    return (
      <div className={styles.wrap}>
        <button
          className={styles.input}
          onClick={(event) => this.onClick(event)}
          type={type}>
          {label}
        </button>
      </div>
    );
  }
}

FormButton.propTypes = {
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  type: PropTypes.oneOf(['submit', 'button', 'reset']).isRequired
};

export default FormButton;
