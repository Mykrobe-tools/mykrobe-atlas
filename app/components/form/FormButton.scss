.wrap {
	composes: wrap from 'styles/form.scss';
}

.input {
	composes: buttonPrimary from 'styles/controls.scss';
	cursor: pointer;
}
