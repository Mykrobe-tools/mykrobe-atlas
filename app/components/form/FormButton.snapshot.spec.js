import React from 'react';
import renderer from 'react-test-renderer';
import FormButton from './FormButton';

describe('FormButton component snapshot', () => {
  const label = 'Lorem ipsum';
  const type = 'submit';
  const onClick = jest.fn();

  it('renders correctly', () => {
    const component = renderer.create(
      <FormButton
        label={label}
        type={type}
        onClick={onClick} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('runs click event', () => {
    const component = renderer.create(
      <FormButton
        label={label}
        type={type}
        onClick={onClick} />
    );
    let tree = component.toJSON();
    const btn = tree.children[0];
    btn.props.onClick();
    expect(onClick).toBeCalled();
  });
});
