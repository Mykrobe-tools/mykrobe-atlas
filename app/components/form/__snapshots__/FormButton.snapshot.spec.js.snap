exports[`FormButton component snapshot renders correctly 1`] = `
<div
  className="wrap">
  <button
    className="input"
    onClick={[Function]}
    type="submit">
    Lorem ipsum
  </button>
</div>
`;
